test/integration/cdi/example-driver/app/kubeletplugin.go

```diff
@@ -0,0 +1,199 @@
+	var possibleResourceHandkeKeys = []string{}
+	for _, key := range keys {
+		possibleResourceHandkeKeys = append(possibleResourceHandkeKeys, "user_"+key)
+		possibleResourceHandkeKeys = append(possibleResourceHandkeKeys, "admin_"+key)
+	}
+	return possibleResourceHandkeKeys
+}
+
+// getJsonFilePath returns the absolute path where json file is/should be.
+func (ex *examplePlugin) getJsonFilePath(claimUID string) string {
+	return filepath.Join(ex.config.cdiDir, ex.config.driverName+"-"+claimUID+".json")
+}
+
+// isKeyValid returns true if the key is valid.
+func (ex *examplePlugin) isKeyValid(key string) bool {
+	for _, k := range ex.resourceHandleKeys {
+		if k == key {
+			return true
+		}
+	}
+	return false
+}
+
+// validateResourceHandle checks if the resourceHandle has only the valid keys that driver writer decided.
+func (ex examplePlugin) validateResourceHandle(rh map[string]string) error {
+	for key := range rh {
+		if !ex.isKeyValid(key) {
+			return fmt.Errorf("invalid key for ResourceHandle")
+		}
+	}
+	return nil
+}
+
+// startPlugin starts servers that are necessary for plugins and registers the plugin with kubelet.
+func (ex examplePlugin) startPlugin() error {
+	klog.Infof("Starting kubelet plugin")
+
+	// create CDI directory if not exists
+	if err := os.MkdirAll(ex.config.cdiDir, os.FileMode(0750)); err != nil {
+		return fmt.Errorf("failed to create directory: %v", err)
+	}
+
+	// create DRA driver socket directory if not exists.
+	if err := os.MkdirAll(filepath.Dir(ex.config.draAddress), 0750); err != nil {
+		return fmt.Errorf("failed to create DRA driver socket directory: %v", err)
+	}
+
+	// Run a grpc server for the driver, that listens on draAddress
+	if err := plugin.StartNonblockingGrpcServer(ex.config.draAddress, &ex); err != nil {
+		return fmt.Errorf("failed to start a nonblocking grpc server: %v", err)
+	}
+
+	// Run a registration server and registers plugin with kubelet
+	if err := plugin.StartRegistrar(ex.config.driverName, ex.config.draAddress, ex.config.pluginRegistrationPath); err != nil {
+		return fmt.Errorf("failed to start registrar: %v", err)
+	}
+
+	return nil
+
+}
+
+func (ex *examplePlugin) NodePrepareResource(ctx context.Context, req *drapbv1.NodePrepareResourceRequest) (*drapbv1.NodePrepareResourceResponse, error) {
+
+	const (
+		deviceName = "exampledevice"
+		cdiVersion = "0.2.0"
+	)
+	kind := ex.config.driverName + "/" + deviceName
+
+	klog.Infof("NodePrepareResource is called: request: %+v", req)
+	// create CDI Files
+
+	// create spec
+	klog.Infof("Creating CDI Files")
+
+	// resourceHandle in the form of map[string]string
+	var decodedResourceHandle map[string]string
+	if err := json.Unmarshal([]byte(req.ResourceHandle), &decodedResourceHandle); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal resourceHandle: %v", err)
+	}
+	// validate if resourceHandle
+	if err := ex.validateResourceHandle(decodedResourceHandle); err != nil {
```

Comment:
I'd skip this for the sake of code simplicity. If resource handle can be unmarshalled into map[string]string we can simply set env vars out of it.